corehq/motech/models.py

```diff
@@ -27,18 +63,24 @@ class ConnectionSettings(models.Model):
     url = models.CharField(max_length=255)
     auth_type = models.CharField(
         max_length=7, null=True, blank=True,
-        choices=(
-            (None, "None"),
-            (BASIC_AUTH, "Basic"),
-            (DIGEST_AUTH, "Digest"),
-            (OAUTH1, "OAuth1"),
-        )
+        choices=((None, "None"),) + AUTH_TYPES
     )
-    username = models.CharField(max_length=255)
-    password = models.CharField(max_length=255)
+    username = models.CharField(max_length=255, blank=True)
+    password = models.CharField(max_length=255, blank=True)
+    api_auth_settings = models.ForeignKey(
+        ApiAuthSettings, null=True, on_delete=models.PROTECT
+    )
+    client_id = models.CharField(max_length=255, null=True, blank=True)
+    client_secret = models.CharField(max_length=255, null=True, blank=True)
```

Comment:
do we ever need both username/password and client_id/client_secret? Could we not overload the username/password fields?

Reply:
Yeah, I was hoping the same thing, and that's how I wrote it first ... until I got to the unit test and discovered the grant type we are using ("resource owner password credentials", aka "password") needs all four. See how we have to [fetch the bearer token here](https://github.com/dimagi/commcare-hq/blob/83a1aa195c980b8439057c60e48f212546ef46bd/corehq/motech/requests.py#L162).